### PR TITLE
Dev/171174 automation framework

### PR DESCRIPTION
**Pull request by shivaMastek**
This PR contains project skeleton for the test automation framework (NUnit using RestSharp).

### CODE REVIEW

**Review by Mayuresh-Gawde — APPROVED** [review]
LGTM!!

**Review by harshal11869 — APPROVED** [review]
Looks Good.

File: tests/UKHO.S100PermitService.API.FunctionalTests/Helpers/TestConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿using Microsoft.Extensions.Configuration;
+
+namespace UKHO.S100PermitService.API.FunctionalTests.Helpers
+{
+    public class TestConfiguration
+    {
+        protected IConfigurationRoot configurationRoot;
+        public PermitServiceApiConfiguration permitServiceConfig = new();
```
Comment (jrippington) [inline tests/UKHO.S100PermitService.API.FunctionalTests/Helpers/TestConfiguration.cs:8]: This should be called PermitServiceConfig if it is public.

File: tests/UKHO.S100PermitService.API.FunctionalTests/Helpers/TestConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+﻿using Microsoft.Extensions.Configuration;
+
+namespace UKHO.S100PermitService.API.FunctionalTests.Helpers
+{
+    public class TestConfiguration
+    {
+        protected IConfigurationRoot configurationRoot;
```
Comment (jrippington) [inline tests/UKHO.S100PermitService.API.FunctionalTests/Helpers/TestConfiguration.cs:7]: Please use the underscore convention for non-public instance members, so _configurationRoot. Could you adjust your editor configuration accordingly?

**Review by jrippington — CHANGES_REQUESTED** [review]
Just a couple of minor changes - casing for publics, and a change for the team editor config as noted please. Looking good!

File: tests/UKHO.S100PermitService.API.FunctionalTests/Helpers/TestConfiguration.cs (hunk 1 above)
Comment (shivaMastek) [reply tests/UKHO.S100PermitService.API.FunctionalTests/Helpers/TestConfiguration.cs:8]: Hi John
Thanks for you comment. As permitServiceConfig is a class object. Therefore we have used camelCase naming convention.

File: tests/UKHO.S100PermitService.API.FunctionalTests/Helpers/TestConfiguration.cs (hunk 2 above)
Comment (shivaMastek) [reply tests/UKHO.S100PermitService.API.FunctionalTests/Helpers/TestConfiguration.cs]: I have made the required change.

File: tests/UKHO.S100PermitService.API.FunctionalTests/Helpers/TestConfiguration.cs (hunk 1 above)
Comment (jrippington) [reply tests/UKHO.S100PermitService.API.FunctionalTests/Helpers/TestConfiguration.cs:8]: Hi, that's not correct - Pascal casing please.

Comment (jrippington) [reply tests/UKHO.S100PermitService.API.FunctionalTests/Helpers/TestConfiguration.cs:8]: Please also provide public get, private set?

**Review by jrippington — CHANGES_REQUESTED** [review]

File: tests/UKHO.S100PermitService.API.FunctionalTests/Helpers/TestConfiguration.cs (hunk 1 above)
Comment (shivaMastek) [reply tests/UKHO.S100PermitService.API.FunctionalTests/Helpers/TestConfiguration.cs]: Hey John
We have made both the changes as per the review comments.

File: tests/UKHO.S100PermitService.API.FunctionalTests/Helpers/TestConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
+﻿using Microsoft.Extensions.Configuration;
+
+namespace UKHO.S100PermitService.API.FunctionalTests.Helpers
+{
+    public class TestConfiguration
+    {
+        protected IConfigurationRoot _configurationRoot;
+        public PermitServiceApiConfiguration PermitServiceConfig { get; private set; }
```
Comment (jrippington) [inline tests/UKHO.S100PermitService.API.FunctionalTests/Helpers/TestConfiguration.cs:8]: Where is this instantiated? Do you need to create a default instance in the ctor()?

File: tests/UKHO.S100PermitService.API.FunctionalTests/Helpers/TestConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+        public class PermitServiceApiConfiguration
```
Comment (jrippington) [inline tests/UKHO.S100PermitService.API.FunctionalTests/Helpers/TestConfiguration.cs:10]: Not sure we need a nested class here, as there is no scoping issue to resolve? Would be clearer as a namespace level declaration.

**Review by jrippington — CHANGES_REQUESTED** [review]
Couple more points, sorry!

File: tests/UKHO.S100PermitService.API.FunctionalTests/Helpers/TestConfiguration.cs (hunk 3 above)
Comment (shivaMastek) [reply tests/UKHO.S100PermitService.API.FunctionalTests/Helpers/TestConfiguration.cs:8]: Sorry, missed pushing this change. Have push the required code in recent commit.

File: tests/UKHO.S100PermitService.API.FunctionalTests/Helpers/TestConfiguration.cs [hunk 5]
```diff
@@ -0,0 +1,20 @@
+﻿using Microsoft.Extensions.Configuration;
+
+namespace UKHO.S100PermitService.API.FunctionalTests.Helpers
```
Comment (jrippington) [inline tests/UKHO.S100PermitService.API.FunctionalTests/Helpers/TestConfiguration.cs:3]: Can we change the namespace to UKHO.S100PermitService.API.FunctionalTests.Configuration please? Avoid using 'helpers' to name things, as it doesn't conform to SOLID principles well.

**Review by jrippington — CHANGES_REQUESTED** [review]
Really sorry, one more thing! My attention keeps getting pulled to other things - last change!

File: tests/UKHO.S100PermitService.API.FunctionalTests/Helpers/TestConfiguration.cs (hunk 5 above)
Comment (shivaMastek) [reply tests/UKHO.S100PermitService.API.FunctionalTests/Helpers/TestConfiguration.cs]: Have renamed the namespace, will follow this practice ahead as well. Thanks!

**Review by jrippington — APPROVED** [review]
Looks good, thanks!

**Pull request merged**